src/Riok.Mapperly/Descriptors/MappingBuilders/DictionaryMappingBuilder.cs

```diff
@@ -38,6 +42,13 @@ public static class DictionaryMappingBuilder
                 dictionaryObjectFactory);
         }
 
+        // if target is an immutable dictionary then use LinqDictionaryMapper
+        if (SymbolEqualityComparer.Default.Equals(ctx.Target.OriginalDefinition, ctx.Types.ImmutableDictionaryT))
```

Comment:
Couldn't we add the `IImmutableDictionary<K, V>` to the check here?

Reply:
Would this change anythng? We'd then have to check if it's an `ImmutableDictionary` or `ImmutableSortedDictionary` to pass the correct collect method name.

I could extract the logic like `TryBuildImmutableLinqMapping` in `EnumerableMappingBuilder`.